sklearn/preprocessing/_data.py

```diff
@@ -2748,8 +2749,30 @@ def _dense_fit(self, X, random_state):
             X = resample(
                 X, replace=False, n_samples=self.subsample, random_state=random_state
             )
+            if sample_weight is not None:
+                sample_weight = sample_weight[: X.shape[0]]
+
+        self.quantiles_ = np.zeros((len(references), n_features))
+
+        for i in range(n_features):
+            col = X[:, i]
+            mask = ~np.isnan(col)
+            col_clean = col[mask]
+
+            if col_clean.size == 0:
+                self.quantiles_[:, i] = np.nan
+                continue
+
+            if sample_weight is not None:
+                weights_clean = sample_weight[mask]
+                self.quantiles_[:, i] = _averaged_weighted_percentile(
+                    col_clean, sample_weight=weights_clean, quantile=references / 100.0
+                )
```

Comment:
Please check the source code of `_averaged_weighted_percentile` and the docstring of `_weighted_percentile` to learn how to use it properly.

In particular:

- it accepts percentile ranks, not quantile ranks;
- it can compute for for a single rank at a time, hence you might need to call this in a list comprehension for all possible ranks passed in the `references` array.

Reply:
Note that `_averaged_weighted_percentile` has now been merged into `_weighted_percentile` via the `average=True` option.